migrate/repo_checks.py

```diff
@@ -0,0 +1,364 @@
+"""
+Run checks Against Repos and correct them if they're missing something.
+
+"""
+from itertools import chain
+from pprint import pformat
+
+import click
+import requests
+from fastcore.net import HTTP4xxClientError, HTTP5xxServerError, HTTP404NotFoundError
+from ghapi.all import GhApi, paged
+
+
+class Check:
+    def __init__(self, api, org, repo):
+        raise NotImplementedError
+
+    def check(self):
+        raise NotImplementedError
+
+    def fix(self):
+        raise NotImplementedError
+
+    def dry_run(self):
+        raise NotImplementedError
+
+
+class RequiredCLACheck(Check):
+    """
+    This class validates the following:
+
+    * Branch Protection is enabled on the default branch.
+    * The CLA Check is a required check.
+
+    If the check fails, the fix function can update the repo
+    so that it has branch protection enabled with the "openedx/cla"
+    check as a required check.
+    """
+
+    def __init__(self, api, org_name, repo_name):
+        self.api = api
+        self.org_name = org_name
+        self.repo_name = repo_name
+
+        self.cla_check = {"context": "openedx/cla", "app_id": -1}
+        self.cla_team = "cla-checker"
+
+        self.has_a_branch_protection_rule = False
+        self.branch_protection_has_required_checks = False
+        self.required_checks_has_cla_required = False
+        self.team_setup_correctly = False
+
+    def check(self):
+        is_required_check = self._check_cla_is_required_check()
```

Comment:
In the longer term, it might be worthwhile to think about decoupling the logic of "how does this check run" from "should this check run on this repo", since the latter is a policy decision that may be very separate from the logic of how to actually execute those checks.

Reply:
One of the things that putting it in the same place makes me wonder off the top of my head is, "What is the result of check() if the repo doesn't pass the check, but it's also not a required check for this repo."

Reply:
FYI, not addressing this in this round but I'm thinking more about this and will probably add a separate function to see if it should run separate from whether or not the check would succeed.  I'll probably try to make that change as I add more checks.